doc/ir-semantics.text

```diff
@@ -2,34 +2,42 @@
 
 x        variables
 
-P ::= Segment  active   ↦ S, (Ds ↦ S)*   program:            active version and others
+P ::= Function main    ↦ F, (Df ↦ F)*    program:            main function and other functions
+F ::= Segment  active  ↦ S, (Ds ↦ S)*    function:           active version and others
 S ::= start ↦ i,          I              segment:            instruction stream with dedicated start
 I ::= (l ↦ i)*                           instruction stream: labeled instructions
 
 Ds ::= Segment  s                  segment declaration
 
+f  function name
 s  optimization version
 l  labels
 
+main      main function
 active    active segment
 start     start label
 
 a  addresses
 
 i ::=    instructions
-| const x = c                   constant variable
-| mut x = c                     mutable variable
-| x <- c                        assignment
+| const x = ce                  constant variable
+| mut x = ce                    mutable variable
+| x <- ce                       assignment
 | branch e l₁ l₂                conditional
 | goto l                        goto
 | print e                       print
+| return e                      return
```

Comment:
So why is `print ce` not allowed? It would be simpler to not make the distinction, but I'm sure there is a rationale -- I thought it would be easier when `print [.]` is already a continuation.